ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -469,6 +469,39 @@ - (void)handleLongPress:(UITapGestureRecognizer *)sender
     }
 }
 
+- (void)mapViewDidFinishLoadingMap:(MGLMapView *)mapView
+{
+    NSDictionary *event = @{ @"target": self.reactTag,
+                             @"src": @{
+                                     @"loaded": @YES
+                                     }
+                             };
+    
+    [_eventDispatcher sendInputEventWithName:@"onFinishLoadingMap" body:event];
+}
+- (void)mapViewWillStartLoadingMap:(MGLMapView *)mapView
+{
+    NSDictionary *event = @{ @"target": self.reactTag,
+                             @"src": @{
+                                     @"loading": @YES
+                                     }
+                             };
+    
+    [_eventDispatcher sendInputEventWithName:@"onStartLoadingMap" body:event];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFailToLocateUserWithError:(NSError *)error
+{
+    NSDictionary *event = @{ @"target": mapView.reactTag,
+                             @"src": @{
+                                     @"code": @(error.code),
```

Comment:
`description` is intended for debugging purposes; you should never compare it to a string to identify a particular condition. The wording could also potentially change from release to release. An NSError object should be identified by its `code`, but that means you may have to export the CLError values as constants, just as you did for style URLs. Also, you should also send along the error’s `localizedDescription`: this is the potentially localized error message that’s suitable for presenting to the user.